Assets/Mirage/Runtime/NetworkSceneManager.cs

```diff
@@ -182,210 +173,468 @@ internal void OnClientChangeScene(string scenePath, SceneOperation sceneOperatio
         /// </summary>
         /// <param name="scenePath">Path of the scene that was just loaded</param>
         /// <param name="sceneOperation">Scene operation that was just  happen</param>
-        internal void OnClientSceneChanged(string scenePath, SceneOperation sceneOperation)
+        internal void OnClientSceneLoadFinished(string scenePath, SceneOperation sceneOperation)
         {
-            if (pendingAdditiveSceneList.Count > 0 && Client && !Client.IsLocalClient)
+            if (ClientPendingAdditiveSceneLoadingList.Count > 0 && Client && !Client.IsLocalClient)
             {
-                ApplyOperationAsync(pendingAdditiveSceneList[0], SceneOperation.LoadAdditive).Forget();
-                pendingAdditiveSceneList.RemoveAt(0);
+                LoadSceneAsync(ClientPendingAdditiveSceneLoadingList[0], new[] {Client.Player}, SceneOperation.LoadAdditive).Forget();
+
+                ClientPendingAdditiveSceneLoadingList.RemoveAt(0);
+
                 return;
             }
 
             //set ready after scene change has completed
-            if (!Client.Player.IsReady)
-                SetClientReady();
+            if (!Client.Player.SceneIsReady)
+                SetSceneIsReady();
 
             //Call event once all scene related actions (subscenes and ready) are done.
-            ClientSceneChanged?.Invoke(scenePath, sceneOperation);
+            ClientFinishedSceneChange?.Invoke(scenePath, sceneOperation);
+        }
+
+        /// <summary>
+        ///     Received message that player is not ready.
+        ///
+        ///     <para>Default implementation is to set player to not ready.</para>
+        /// </summary>
+        /// <param name="player">The player that is currently not ready.</param>
+        /// <param name="message">The message data coming in.</param>
+        public virtual void ClientNotReadyMessage(INetworkPlayer player, NotReadyMessage message)
```

Comment:
Should this method be public?

Reply:
Same as before do we need to make it protected.